packages/network-console-shared/src/file_io/native/environment-format.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+    IEnvironmentFormat,
+    IEnvironmentContainerAdapter,
+} from '../interfaces';
+import { EnvironmentContainerAdapter } from './environment-container-adapter';
+import { INetConsoleParameter } from '../../net/net-console-http';
+
+export interface INCNativeEnvironmentFile {
+    meta: {
+        networkConsoleEnvironmentVersion: string;
+    };
+    name: string;
+    environments: INCNativeEnvironment[];
+}
+
+export interface INCNativeEnvironment {
+    name: string;
+    variables: INetConsoleParameter[];
+}
+
+export class EnvironmentFormat implements IEnvironmentFormat {
+    public readonly formatId = 'nc-native-env';
+    public readonly canWrite = true;
+
+    private static _nextNewEnvironmentId = 0;
+
+    constructor() {}
+
+    async createEnvironmentContainer(name: string): Promise<IEnvironmentContainerAdapter> {
+        const DEFAULT_ENVIRONMENT: INCNativeEnvironmentFile = {
+            meta: {
+                networkConsoleEnvironmentVersion: '0.9.2-preview',

```

Comment:
a constant maybe?

Reply:
I think this will be a longer-term issue for how we solve this for differentiating versions (e.g., if there is a new schema in the future). For the moment, because it's also in json files, I don't want to solve it this way (preferring instead to be able to do a grep).